packages/plugin-avatar/test/unit/spec/avatar.js

```diff
@@ -0,0 +1,440 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ */
+import {assert} from '@ciscospark/test-helper-chai';
+import Avatar from '../../';
+import User from '@ciscospark/plugin-user';
+import MockSpark from '@ciscospark/test-helper-mock-spark';
+import sinon from '@ciscospark/test-helper-sinon';
+
+
+describe(`Services`, () => {
+  describe(`Avatar`, () => {
+    let avatar;
+    let spark;
+
+    beforeEach(() => {
+      spark = new MockSpark({
+        children: {
+          avatar: Avatar,
+          user: User
+        }
+      });
+      avatar = spark.avatar;
+      avatar.config.cacheExpiration = 60 * 60;
+      avatar.config.defaultAvatarSize = 80;
+    });
+
+    describe(`#retrieveAvatarUrl`, () => {
+      it(`requires a user identifying object`, () => {
+        return Promise.all([
+          assert.isRejected(avatar.retrieveAvatarUrl(), `\`user\` is a required parameter`)
+        ]);
+      });
+
+      describe(`request`, () => {
+        it(`retrieves an avatar url`, () => {
+          spark.request = sinon.stub().returns(Promise.resolve({
+            body: {
+              '88888888-4444-4444-4444-aaaaaaaaaaa0': {
+                80: {
+                  size: 80,
+                  url: `https://example.com/88888888-4444-4444-4444-aaaaaaaaaaa0`
+                }
+              }
+            },
+            statusCode: 200,
+            options: {
+              ids: [
+                `88888888-4444-4444-4444-aaaaaaaaaaa0`
+              ],
+              body: [
+                {
+                  uuid: `88888888-4444-4444-4444-aaaaaaaaaaa0`,
+                  sizes: [80]
+                }
+              ]
+            }
+          }));
+
+          return assert.becomes(avatar.retrieveAvatarUrl(`88888888-4444-4444-4444-aaaaaaaaaaa0`), `https://example.com/88888888-4444-4444-4444-aaaaaaaaaaa0`);
```

Comment:
fixed in latest commit, wasn't setting batch config vars correctly, not defining BatchWait etc.